### DISCUSSION TITLE
IndexError: arrays used as indices must be of integer (or boolean) type

### DISCUSSION THREAD

**Discussion by luobotaxinghu** [openai/whisper#1293] [Q&A] [answered]
Large model will succeed 99% of my audio files, but 1% of audio files, will encounter the following problems:
`
> File "/home/recsys/Voice2text/task/whisper_voice2text.py", line 115, in parse
    result = self.model.transcribe(audio=path,
             │    │     │                └ './data/voice0/293360_24372204_644a89ad218b849877aaada8_25633001_24.m4a'
             │    │     └ <function transcribe at 0x7f167d11b310>
             │    └ Whisper(
             │        (encoder): AudioEncoder(
             │          (conv1): Conv1d(80, 1280, kernel_size=(3,), stride=(1,), padding=(1,))
             │          (conv2): C...
             └ <task.whisper_voice2text.WhisperVoice2Text object at 0x7f173e666b50>

  File "/home/anaconda/envs/whisper/lib/python3.9/site-packages/whisper/transcribe.py", line 316, in transcribe
    add_word_timestamps(
    └ <function add_word_timestamps at 0x7f16773645e0>
  File "/home/anaconda/envs/whisper/lib/python3.9/site-packages/whisper/timing.py", line 294, in add_word_timestamps
    alignment = find_alignment(model, tokenizer, text_tokens, mel, num_frames, **kwargs)
                │              │      │          │            │    │             └ {}
                │              │      │          │            │    └ 3000
                │              │      │          │            └ tensor([[ 0.9380,  0.9106,  0.8604,  ..., -0.1737,  0.0109, -0.1735],
                │              │      │          │                      [ 1.1279,  1.0703,  1.0215,  ..., -0.0634, -0.1...
                │              │      │          └ []
                │              │      └ Tokenizer(tokenizer=GPT2TokenizerFast(name_or_path='/home/anaconda/envs/whisper/lib/python3.9/site-packages/whisper/assets/mu...
                │              └ Whisper(
                │                  (encoder): AudioEncoder(
                │                    (conv1): Conv1d(80, 1280, kernel_size=(3,), stride=(1,), padding=(1,))
                │                    (conv2): C...
                └ <function find_alignment at 0x7f16773641f0>
  File "/home/anaconda/envs/whisper/lib/python3.9/site-packages/whisper/timing.py", line 217, in find_alignment
    start_times = jump_times[word_boundaries[:-1]]
                  │          └ array([0.])
                  └ array([0.])

IndexError: arrays used as indices must be of integer (or boolean) type
`
my env:
python 3.9.16

ffmpeg-python               0.2.0
nvidia-cublas-cu11          11.10.3.66
nvidia-cuda-nvrtc-cu11      11.7.99
nvidia-cuda-runtime-cu11    11.7.99
nvidia-cudnn-cu11           8.5.0.96
openai-whisper              20230308
tokenizers                  0.13.2
tomli                       2.0.1
torch                       1.12.1+cu113
torchaudio                  0.12.1+cu113
torchvision                 0.13.1+cu113
transformers                4.26.1

**Comment by jongwook** [comment, marked as answer]
Hi, thanks for reporting this - #1317 should fix this.
